### ISSUE TITLE
Change cmb2 label for fields group for rest api

### ISSUE THREAD

**Issue by Mte90** [CMB2/CMB2#1239]
## Expected Behavior:
When you open a rest api page you see the fields under the group of `cmb2` but this should be possible to change it based on the needs.

## Actual Behavior:
Right now is not possible.

## Possible Solution
Add a filter there https://github.com/CMB2/CMB2/blob/develop/includes/rest-api/CMB2_REST.php#L237

**Comment by jtsternberg** [comment]
To change what? the fields that show?

**Comment by Mte90** [comment]
![immagine](https://user-images.githubusercontent.com/403283/54206432-91b9ac00-44d8-11e9-9718-06f4bc2c18f2.png)
As you can see here cmb2 groups all the fields in that group but for a public endpoint is not beautiful and clear what means.
If it is possible to change will be better :-)

**Comment by jtsternberg** [comment]
Ah! you mean change the `cmb2` prefix. That makes sense, but it's also problematic as it could break other plugins or theme which is depending on the response looking that way. This would need some thought.

**Comment by Mte90** [comment]
I see that is not easy to do that change because the group name is used to check in the rest of that class to show the values.
Probably add that feature means rewrite all this class to add support for a big astraction (and avoid conflicts with other plugins).

Maybe we can add a value in the metabox that is used to define the group and change the flow of the various methods.

**Comment by jtsternberg** [comment]
Yes, that could definitely work. We could use the [foreach loop](https://github.com/CMB2/CMB2/blob/associated-objects-field/includes/rest-api/CMB2_REST.php#L176) to group all the boxes into their defined rest slug. It will take a bit of work to change the logic here, but could definitely be done.

**Comment by Mte90** [comment]
I am starting to experimenting with https://github.com/Mte90/CMB2/commit/e664f73ce780a8069c02632473a80bff139b36a9
It is not perfect because should ignore the metabox of other post types anyway and is not rewriting the flow.

**Comment by Mte90** [comment]
With this other commit https://github.com/Mte90/CMB2/commit/125848c0e562620ebf5add566faf7170b030493b seems that now is altro backward compatible and don't conflicts with other post type or metabox.
I did that without rewrite everything as suggested so I am not sure if it is the case to do a pr for it.

How works:

* Load the metabox
* Check for a parameter in the metabox that is the slug for the api
* Owerwrite for that metabox instance the slug

In this way don't create conflicts and we don't need to refactor the class.
